cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_collaborators.spec.js

```diff
@@ -0,0 +1,208 @@
+            write: { users: ['%me%'] },
+          },
+        },
+      }).then((value) => (workspaceId = value));
+    });
+
+    beforeEach(() => {
+      // Visit workspace collaborators page
+      miscUtils.visitPage(`w/${workspaceId}/app/workspace_collaborators`);
+    });
+
+    after(() => {
+      cy.deleteWorkspaceById(workspaceId);
+    });
+
+    it('should successfully load the page', () => {
+      if (Cypress.env('SAVED_OBJECTS_PERMISSION_ENABLED')) {
+        cy.contains('Collaborators', { timeout: 60000 }).should('be.visible');
+      }
+    });
+
+    if (
+      Cypress.env('SAVED_OBJECTS_PERMISSION_ENABLED') &&
+      Cypress.env('SECURITY_ENABLED')
+    ) {
+      describe('Update collaborators', () => {
+        it('should add user and group collaborators successfully', () => {
+          //Add user
+          cy.getElementByTestId('add-collaborator-popover').click();
+          cy.get('button[id="user"]').click();
+          cy.contains('Add Users').should('be.visible');
+          cy.getElementByTestId('workspaceCollaboratorIdInput-0').type(
+            'read_user'
+          );
+          cy.get('button[type="submit"]')
+            .contains('span', 'Add collaborators')
+            .click();
+
+          //Add group
+          cy.getElementByTestId('add-collaborator-popover').click();
+          cy.get('button[id="group"]').click();
+          cy.contains('Add Groups').should('be.visible');
+          cy.getElementByTestId('workspaceCollaboratorIdInput-0').type(
+            'admin_group'
+          );
+          cy.get('span[title="Admin"]').click();
+          cy.get('button[type="submit"]')
+            .contains('span', 'Add collaborators')
+            .click();
+
+          cy.wait(2000); // Intentional Wait
+
+          cy.get('table')
+            .contains('td', 'read_user')
+            .parent()
+            .within(() => {
+              cy.get('td').eq(3).contains('Read only');
+            });
+
+          cy.get('table')
+            .contains('td', 'admin_group')
+            .parent()
+            .within(() => {
+              cy.get('td').eq(3).contains('Admin');
+            });
+          const expectedWorkspace = {
+            name: workspaceName,
+            features: ['use-case-observability'],
+            description: 'test_description',
+            permissions: {
+              library_write: {
+                users: [`${Cypress.env('username')}`],
+                groups: ['admin_group'],
+              },
+              write: {
+                users: [`${Cypress.env('username')}`],
+                groups: ['admin_group'],
+              },
+              library_read: {
+                users: ['read_user'],
+              },
+              read: {
+                users: ['read_user'],
+              },
+            },
+          };
+          cy.checkWorkspace(workspaceId, expectedWorkspace);
+        });
+
+        it('should change access level successfully', () => {
```

Comment:
Seems this it depends on `should add user and group collaborators successfully`, can we decouple these integration tests?